### PR TITLE
Doc updates and deploy

### PR DESCRIPTION
**Pull request by acpaquette**
Updates the docs to be a little more cohesive, also adds the job to deploy to AWS when merged in main

### CODE REVIEW

File: .github/workflows/aws_doc_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: aws-doc-deploy
+
+on:
+  workflow_run:
+    workflows: ["Build-Docs"]
```
Comment (jlaura) [inline .github/workflows/aws_doc_deploy.yml:5]: This is pretty sweet. It uses the name from any other run CI step as the key and if that completes, it fires? Nice.

File: .github/workflows/aws_doc_deploy.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      
+      - name: Upload to S3
+        working-directory: docs/public
+        run: |
+          cd docs
+          aws s3 sync --delete public s3://asc-docs/usgscsm
```
Comment (jlaura) [inline .github/workflows/aws_doc_deploy.yml:37]: One thing that might make sense is to use a sub "directory" here. Like s3://asc-docs/software-docs or something? I am thinking of the use case where /tutorials has a site with tutorials, etc.

File: docs/source/user/isd_generation.rst [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 Creating and testing an ISD
-===
+===========================
 
 This tutorial assumes that the user has created a conda environment with ale, usgscsm,
 and csmapi installed.
 
 Generating an ISD with ALE
------------------
+--------------------------
```
Comment (jlaura) [inline docs/source/user/isd_generation.rst:10]: I ❤️ sphinx and it's wonky warnings....

**Review by jlaura — APPROVED** [review]

File: .github/workflows/aws_doc_deploy.yml (hunk 2 above)
Comment (acpaquette) [reply .github/workflows/aws_doc_deploy.yml:37]: Something to consider as the provided documentation matures!

File: .github/workflows/aws_doc_deploy.yml (hunk 1 above)
Comment (acpaquette) [reply .github/workflows/aws_doc_deploy.yml:5]: Ideally, we will see when it gets merged

**Pull request merged**